server/spec/owner_resource_spec.rb

```diff
@@ -690,6 +702,59 @@
     # and only include one instance of each unique value.
     res["systemPurposeAttributes"]["roles"].length.should == 1
   end
+
```

Comment:
> Moved some of the access related tests from Junit to Rspec.
> Junits did not invoke the VerifyAuthorizationFilter
> when the endpoint is ported into OpenAPI.

Having some doubts about these changes of converting Junits to functional Spec tests. Since we are making changes with ``Principle`` which is necessary to port resource class to OpenAPI, not sure if we are tampering with older functionality unknowingly. 
Any thoughts ? @Ceiu @nikosmoum @Januson

Reply:
I agree. Rather than moving tests to RSpec we should have proper integration tests for our filters and other RestEasy stuff.

I would propose to create a card to create said tests and instead of deleting the broken tests to disable them with a card tag.

Reply:
I'm a bit late to respond here, but yeah... let's not move JUnit tests to rspec if we can avoid it. Last time we discussed it, the long-term plan is to get rid of rspec (which is a large part of why we're doing the switch to OpenAPI stuff to begin with), so it'll just create more work for us down the line.

Reply:
I agree that whenever possible, we should have unit tests instead of spec tests. Spec tests take much longer to run, so if a functionality can be tested at the unit test level, we should keep it at that.

Reply:
Agreed. After more digging into this, it came down to be a **Guice AOP [issue](https://github.com/google/guice/issues/201).** 
Currently, we override the methods in resource classes from the interface. So few of the annotation in the resource class method is `@Override`. And other HTTP Annotations are in API interface methods. So in `TestingModules`, we bind authorization interceptor, we use `HttpMethodMatcher`. But It does not find the superclass HTTP annotations and fails to invoke the authorization filter for each method.

To overcome this particular scenario, I have added a code to explicitly add interceptor for OwnerResource :
` bindInterceptor(Matchers.subclassesOf(OwnerResource.class),
                Matchers.any(), authMethodInterceptor);`
This will simply work for any method present in OwnerResource. And it serves the current purpose. Also, we need to look again when we port `ConsumerResource`.